mmv1/products/networkconnectivity/InternalRanges.yaml

```diff
@@ -0,0 +1,151 @@
+    vars:
+      internal_range_name: "overlap-range"
+      network_name: "internal-ranges"
+properties:
+  - !ruby/object:Api::Type::String
+    name: 'name'
+    required: true
+    immutable: true
+    url_param_only: true
+    description: |
+      The name of the policy based route.
+  - !ruby/object:Api::Type::KeyValueLabels
+    name: 'labels'
+    description: |
+      User-defined labels.
+  - !ruby/object:Api::Type::String
+    name: 'description'
+    description: |
+      An optional description of this resource.
+  - !ruby/object:Api::Type::String
+    name: 'ipCidrRange'
+    description: |
+      The IP range that this internal range defines.
+  - !ruby/object:Api::Type::ResourceRef
+    name: 'network'
+    resource: 'Network'
+    imports: 'selfLink'
+    required: true
+    description: |
+      Fully-qualified URL of the network that this route applies to, for example: projects/my-project/global/networks/my-network.
+  - !ruby/object:Api::Type::Enum
+    name: 'usage'
+    required: true
+    description: |
+      The type of usage set for this InternalRange.
+    values:
+      - :FOR_VPC
+      - :EXTERNAL_TO_VPC
+  - !ruby/object:Api::Type::Enum
+    name: 'peering'
+    immutable: true
+    required: true
+    description: |
+      The type of peering set for this internal range.
+    values:
+      - :FOR_SELF
+      - :FOR_PEER
+      - :NOT_SHARED
+  - !ruby/object:Api::Type::Integer
+    name: 'prefixLength'
+    description: |
+      An alternate to ipCidrRange. Can be set when trying to create a reservation that automatically finds a free range of the given size.
+      If both ipCidrRange and prefixLength are set, there is an error if the range sizes do not match. Can also be used during updates to change the range size.
+  - !ruby/object:Api::Type::Array
+    name: 'targetCidrRange'
+    item_type: Api::Type::String
+    description: |
+      Optional. Can be set to narrow down or pick a different address space while searching for a free range.
+      If not set, defaults to the "10.0.0.0/8" address space. This can be used to search in other rfc-1918 address spaces like "172.16.0.0/12" and "192.168.0.0/16" or non-rfc-1918 address spaces used in the VPC.
+  - !ruby/object:Api::Type::Array
+    name: 'users'
+    item_type: Api::Type::String
+    output: true
+    description: |
+      Output only. The list of resources that refer to this internal range.
+      Resources that use the internal range for their range allocation are referred to as users of the range.
+      Other resources mark themselves as users while doing so by creating a reference to this internal range. Having a user, based on this reference, prevents deletion of the internal range referred to. Can be empty.
+  - !ruby/object:Api::Type::Array
+    name: 'overlaps'
+    description: |
+      Optional. Types of resources that are allowed to overlap with the current internal range.
+    item_type: !ruby/object:Api::Type::Enum
+      name: 'undefined'
```

Comment:
note to self: This `name` value never gets used for an array of enums, so this is reasonable.